cmd/antrea-agent/options.go

```diff
@@ -125,7 +126,14 @@ func (o *Options) validate(args []string) error {
 	}
 
 	if encapMode.SupportsNoEncap() {
-		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		// When using NoEncap traffic mode without AntreaProxy, Pod-to-Service traffic is handled by kube-proxy
+		// (iptables/ipvs) in the root netns. If the Endpoint is not local the DNATed traffic will be output to
+		// the physical network directly without going back to OVS for Egress NetworkPolicy enforcement, which
+		// breaks basic security functionality. Therefore, we usually do not allow the NoEncap traffic mode without
+		// AntreaProxy. But one can bypass this check and force this feature combination to be allowed, by defining
+		// the ALLOW_NO_ENCAP_WITHOUT_ANTREA_PROXY environment variable and setting it to true. This may lead to
+		// better performance when using NoEncap if Egress NetworkPolicy enforcement is not required.
+		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) && !env.GetAllowNoEncapWithoutAntreaProxy() {
 			return fmt.Errorf("TrafficEncapMode %s requires AntreaProxy to be enabled", o.config.TrafficEncapMode)
 		}
```

Comment:
sorry for not mentioning it earlier, but I think we agreed last time (original PR) that we would log an `Info` message if AntreaProxy is disabled even if `ALLOW_NO_ENCAP_WITHOUT_ANTREA_PROXY` is set to true:

```go
if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
    if env.GetAllowNoEncapWithoutAntreaProxy() {
        klog.InfoS("Disabling AntreaProxy in noEncap mode will prevent Egress NetworkPolicy rules from being enforced correctly")
    } else {
        return fmt.Errorf("TrafficEncapMode %s requires AntreaProxy to be enabled", o.config.TrafficEncapMode)
    }
}
```

Reply:
Updated